### PR TITLE
TST: new set of tests to the Flight class

### PR DESCRIPTION
**Pull request by Gui-FernandesBR**
## Pull request type

Please check the type of change your PR introduces:

- [x] Other (please describe): Unit tests

## Pull request checklist

Please check if your PR fulfills the following requirements, depending on the type of PR:

- Code base additions (for bug fixes / features):

  - [x] Tests for the changes have been added
  - [x] Docs have been reviewed and added / updated if needed
  - [x] Lint (`black rocketpy`) has passed locally and any fixes were made
  - [x] All tests (`pytest --runslow`) have passed locally

## What is the current behavior?
- The current tests covering the Flight class barely tests the results of the class. Instead, it only check if the class is running without errors. This is a kind of poor testing as it does not give us enough faith on the results if we modify the Flight Class.
- What I propose is for us to set tests checking numeric outputs from the class, thus ensuring that we will always have a way to copare the new flight simulations against the ones 

## What is the new behavior?
- (1) For some specific methods, now we have tests checking if the class is returning the expected values, with those being the results extracted from the simulation as it is today.
- (2) On other hand, some tests are checking if the class is returning the correct value compared against the exact expected value (calculated by hand).
- I documented every test so we can easily differentiate (1) and (2)
- I separated the tests by logic, this way I believe is easier for us to debug the code in the future. Specifically, you're gonna see some very similar methods at the bottom of the test_flight.py file. ButI throught it would be worse to have everything being test under the same function.
- As a painpoint of making more narrow tests like this, we probably will need to change te expected values of several method if we find a serious bug in the code. (but we don't think there's space for such error in the current state of the code, right?)

## Does this introduce a breaking change?

- [x] No

## Other information
- Please review the #407 first, and then we can change the head branch to `beta/v1.0.0`
- Being 100% transparent here: I created all these tests because I want to change the way we calculate R1, R2, R3, M1, M2 and M3 arguments, and therefore I'd love to have better tests to support my changes in a future PR. The F[light class needs an overhaul](https://github.com/RocketPy-Team/RocketPy/issues/276), so I think more robust tests are coming one way or another

### CODE REVIEW

**Comment by Gui-FernandesBR** [comment]
And of course it didn't work out of windows system (great!!)

I'm converting this to draft, temporarily

File: tests/test_flight.py
Comment (Gui-FernandesBR) [reply tests/test_flight.py:642]: I honestly didn't understand what you said, sorry. Could you re-sentence it please?

File: tests/test_flight.py
Comment (Gui-FernandesBR) [reply tests/test_flight.py:808]: nice catch, lemme see what I can do

File: tests/test_flight.py
Comment (Gui-FernandesBR) [reply tests/test_flight.py:703]: I believe `robust_flight` is not the best name because I think in the future we could try adding other rockets beyond calisto. That's why I wanted to preserve the names

File: tests/conftest.py
Comment (Gui-FernandesBR) [inline tests/conftest.py:1]: The conftest.py file is becoming larger everyday.
PErhaps we should separate the fixtures into different files and than importing evrything in the conftest.py file, just like here: https://stackoverflow.com/a/65035367/19971645

File: tests/test_flight.py
Comment (Gui-FernandesBR) [reply tests/test_flight.py:34]: You're right, it was a kinda redundant. I didn't pay enough attention so I created this function without knowing that the assert operator already handles with labels

File: tests/test_flight.py
Comment (Gui-FernandesBR) [reply tests/test_flight.py:703]: Besides my last comment, I have two more reasons for keeping the
* Not the end of the world using the name "test", as long as you clearly define what this test is (test = xxxx)
* Hopefully one day this test can be used for more than one flight object, like in a parametrize fixture scheme.

An alternative, though, is to use the simple name "flight".

I'm against dropping the "calisto" name from the variable. More rockets may be tested in the future.

File: tests/test_flight.py
Comment (Gui-FernandesBR) [reply tests/test_flight.py:808]: I just wanted to ensure that we could easily track which of the parametrized solutions is breaking the code.
If the verbose options of pytest already covers it, I'd be satisfied already.

Comment (phmbressan) [reply tests/test_flight.py]: I have thought a bit further on the matter and decided to keep the labeling in some tests to make troubleshooting easier. The changes can be seem on commit [e1a60f3](https://github.com/RocketPy-Team/RocketPy/pull/408/commits/e1a60f389deb2b8df243cba6822ef5e9be18c577).

File: tests/test_flight.py
Comment (phmbressan) [reply tests/test_flight.py]: Solved by [7cd4241](https://github.com/RocketPy-Team/RocketPy/pull/408/commits/7cd424144810d9fa405dc292694055d205fc1fc3).

File: tests/conftest.py
Comment (Gui-FernandesBR) [reply tests/conftest.py:1]: New issue created for this, see #447

**Review by Gui-FernandesBR — COMMENTED** [review]
Merging the develop branch into this one actually solved the problem that was blocking this PR.
So I believe we are good to go!

I can't merge it bc we need an approval first. Asking for help here.

**Review by phmbressan — APPROVED** [review]
Good. I have updated some old docstrings in my latest commits.

Before merging, I would only double check if the numbers of the new tests are still valid (the ones that come from sim results) since there was a gap of time between this PR being implmented and now.

**Pull request merged**
